lenny/routes/api.py

```diff
@@ -57,17 +58,27 @@ async def get_opds(request: Request, offset: Optional[int]=None, limit: Optional
 
 # Redirect to the Thorium Web Reader
 @router.get("/items/{book_id}/read")
-async def redirect_reader(book_id: str, format: str = "epub"):
-    if not LennyAPI.auth_check(book_id):
-        HTTPException(status_code=400, detail="Unauthorized request")
+async def redirect_reader(request: Request,book_id: str, format: str = "epub"):
+    email = request.cookies.get("email")
+    logged_in = request.cookies.get("logged_in") == "true"
```

Comment:
Let's use https://pypi.org/project/itsdangerous/ as per your suggestion;

We can use their email and our LENNY_SEED for example